src/Orleans.Core/Messaging/MessageSerializer.cs

```diff
@@ -131,13 +128,42 @@ private void ReadBodyObject<TInput>(Message message, ref Reader<TInput> reader)
         {
             var field = reader.ReadFieldHeader();
 
-            var bodyCodec = _codecProvider.GetCodec(field.FieldType);
-            message.BodyObject = bodyCodec.ReadValue(ref reader, field);
+            if (message.Result == ResponseTypes.Success)
+            {
+                message.Result = ResponseTypes.None; // reset raw response indicator
```

Comment:
I'll merge this as-is. In a subsequent PR, I'd like us to have some more explanatory comments about how the ResponseTypes affect body encoding.